java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java

```diff
@@ -151,14 +152,22 @@ static Type signatureTypeToType(SignatureAttribute.Type signatureType, TypeSolve
             return new TypeVariable(typeParameterDeclaration);
         } else if (signatureType instanceof SignatureAttribute.BaseType) {
             SignatureAttribute.BaseType baseType = (SignatureAttribute.BaseType)signatureType;
-            return PrimitiveType.byName(baseType.toString());
+            if (baseType.toString().equals("void")) {
+                return VoidType.INSTANCE;
+            } else {
+                return PrimitiveType.byName(baseType.toString());
+            }
         } else {
             throw new RuntimeException(signatureType.getClass().getCanonicalName());
         }
     }
 
     private static String internalNameToCanonicalName(String typeName) {
-        return typeName.replaceAll("\\$", ".");
+        String canonicalName = typeName.replaceAll("\\$", ".");
+        if (canonicalName.contains("<")) {
```

Comment:
right. However would that be an internal name? Maybe not, and this should be done before calling this method, so that it receives a valid internal name

Reply:
good point!